lib/dao/payment_dao.dart

```diff
@@ -103,4 +105,194 @@ class PaymentDao {
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'title = ? AND autoCategorizationEnabled = ?',
+      whereArgs: [title, true], // Assuming you want to filter for true
+    );
+    if (maps.isNotEmpty) {
+
+      // Returns the Index of Applied Category
+      return (maps.first)['category']-1;
+    }
+    return 9;
+  }
+
+  Future<List<Map<String,dynamic>>?> findPaymentsWithMissingCategory() async {
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'category = ?',
+      whereArgs: [9], // Assuming you want to filter for true
+    );
+    if (maps.isNotEmpty) {
+      return maps;
+    }
+    return null;
+  }
+
+  Future<Map<String,dynamic>?> searchForSameTitle(String title,int id) async{
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'title = ? AND id != ?',
+      whereArgs: [title,id], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps.first;
+    }
+
+    return null;
+  }
+
+
+  Future<List<Map<String,dynamic>>?> getAllMiscellanous() async{
+
+    List<Category> allCategories = await CategoryDao().find();
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'category = ?',
+      whereArgs: [10], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps;
+    }
+
+    return null;
+  }
+
+
+  Future<int> updateMiscellanousCategories() async{
+
+    int count = 0;
+    CategoryDao categoryDao = CategoryDao();
+    AccountDao accountDao = AccountDao();
+    List<Map<String,dynamic>> allMisPayments = await getAllMiscellanous() ?? [];
+    if(allMisPayments.isNotEmpty){
+
+      for(Map<String,dynamic> misPayment in allMisPayments){
+        Map<String,dynamic> gotPayment = await searchForSameTitle(misPayment['title'], misPayment['id']) ?? {};
+        if(gotPayment.isNotEmpty){
+          Category? cat = await categoryDao.findCategoryById(gotPayment['category']);
+          Account? acc = await accountDao.findCategoryById(misPayment['account']);
+          final updatedMap = {"id": misPayment['id'], "title": misPayment['title'], "description": misPayment['description'], "account": acc!.toJson(), "category": cat!.toJson(), "amount": misPayment['account'].toDouble(), "type": misPayment['type'], "datetime": misPayment['datetime'], "autoCategorizationEnabled": misPayment['autoCategorizationEnabled'] == 1 ? true : false};
+          await upsert(Payment.fromJson(updatedMap),);
+          count++;
+        }
+      }
+
+    }
+    return count;
+  }
+
+  Future<int> categorizeUsingRules()async{
+
+    CategoryDao categoryDao = CategoryDao();
+    AccountDao accountDao = AccountDao();
+    final db = await getDBInstance();
+    List<String> titles = [
```

Comment:
You are using if conditions like title == 'toll charges'.
What will you do if there are hundreds or there is an update in the list?

Better solution:
As we are trying to find the category, you can use a map like this:
someDataName = [
"toll charges": "Transportation",
"BookMyShow": "Recreation & Entertainment",
]

For adding user rules, you can add like this:
someDataName.add({title:category}) in the loop.

Reply:
With the said apprach we do not need if conditions.
You can directly get catergory by just calling someDataName[title]. And then get the category from the categoryDao using it.